app/src/main/java/com/github/polybooks/core/database/DatabaseAPI.kt

```diff
@@ -0,0 +1,176 @@
+package com.github.polybooks.core.database
+
+import com.github.polybooks.core.*
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Provides the API for a database. Allows to retrieve information in the database
+ * about Books, Sales.
+ * */
+interface Database {
+
+    /**
+     * Creates a new query for Books. It originally matches all books.
+     * */
+    fun queryBooks() : BookQuery
+
+    /**
+     * Create a new query for Sales. It originally matches all sales.
+     * */
+    fun querySales() : SaleQuery
+
+    //TODO we could or could not include those functions (and more) in the interface. Should we?
```

Comment:
We should be able to display all books and all sales to the user, so I would guess that having them in the interface is useful and improve modularity